File: src/main/java/me/shedaniel/linkie/CommandBase.java
package me.shedaniel.linkie;

import discord4j.core.event.domain.message.MessageCreateEvent;
import discord4j.core.object.entity.Member;
import discord4j.core.object.entity.MessageChannel;

import java.util.concurrent.ExecutionException;
import java.util.concurrent.ScheduledExecutorService;

public interface CommandBase {
    
    void execute(ScheduledExecutorService service, MessageCreateEvent event, Member author, String cmd, String[] args, MessageChannel channel)
            throws ExecutionException, InterruptedException;
    
}

File: src/main/kotlin/me/shedaniel/linkie/CommandBase.kt
package me.shedaniel.linkie

import discord4j.core.`object`.entity.Member
import discord4j.core.`object`.entity.MessageChannel
import discord4j.core.event.domain.message.MessageCreateEvent

interface CommandBase {

    @Throws(Exception::class)
    fun execute(event: MessageCreateEvent, author: Member, cmd: String, args: Array<String>, channel: MessageChannel)

    fun getName(): String? = null
    fun getDescription(): String? = null

}